src/nvim/tui/input.c

```diff
@@ -344,6 +345,31 @@ static void tk_getkeys(TermInput *input, bool force)
       forward_modified_utf8(input, &key);
     } else if (key.type == TERMKEY_TYPE_MOUSE) {
       forward_mouse_event(input, &key);
+    } else if (key.type == TERMKEY_TYPE_UNKNOWN_CSI) {
+      // There is no specified limit on the number of parameters a CSI sequence can contain, so just
+      // allocate enough space for a large upper bound
+      long args[16];
+      size_t nargs = 16;
+      unsigned long cmd;
+      if (termkey_interpret_csi(input->tk, &key, args, &nargs, &cmd) == TERMKEY_RES_KEY) {
+        uint8_t intermediate = (cmd >> 16) & 0xFF;
+        uint8_t initial = (cmd >> 8) & 0xFF;
+        uint8_t command = cmd & 0xFF;
+
+        // Currently unused
+        (void)intermediate;
+
+        if (input->waiting_for_extkeys_response > 0) {
+          if (initial == '?' && command == 'u') {
+            input->supports_kitty_protocol = true;
```

Comment:
If the following changes are made in conjunction with the changes to [r856873474](https://github.com/neovim/neovim/pull/18181/files#r856873474), the two variables `supports_kitty_protocol` and `supports_modify_other_keys` should not be needed.


```C
data->unibi_ext.enable_extended_keys = (int)unibi_add_ext_str(ut, "ext.enable_extended_keys",
                                                              "\x1b[>1u");
data->unibi_ext.disable_extended_keys = (int)unibi_add_ext_str(ut,
                                                               "ext.disable_extended_keys",
                                                               "\x1b[<1u");
```

Reply:
We still need this flag because we need somewhere to store the result when decoding the response from the terminal.